File: frontend/src/types/transaction.ts
/**
 * バックエンドの TransactionCreate スキーマ（backend/app/schemas/transaction.py）に対応するリクエスト型。
 */
export interface TransactionCreatePayload {
  date: string;
  store_name: string;
  amount: number;
  category: string;
  payment_method: string;
  memo: string | null;
  receipt_image: string | null;
}

/**
 * バックエンドの TransactionUpdate スキーマに対応するリクエスト型。
 * 現状 TransactionCreate と同じ形なので型エイリアスとして定義する。
 */
export type TransactionUpdatePayload = TransactionCreatePayload;

/**
 * バックエンドの TransactionRead スキーマに対応するレスポンス型。
 */
export interface TransactionApiResponse {
  id: number;
  date: string;
  store_name: string;
  amount: number;
  category: string;
  payment_method: string;
  memo: string | null;
  receipt_image: string | null;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/transaction.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field


class TransactionBase(BaseModel):
    date: date
    store_name: str = Field(..., min_length=1, max_length=255)
    amount: int = Field(..., gt=0)
    category: str = Field(..., min_length=1, max_length=50)
    payment_method: str = Field(..., min_length=1, max_length=50)
    memo: str | None = Field(default=None, max_length=1000)
    # 画像本体ではなく保存先パスのみ（例: uploads/receipts/20260716_001.jpg）
    receipt_image: str | None = Field(default=None, max_length=500)


class TransactionCreate(TransactionBase):
    pass


class TransactionUpdate(TransactionBase):
    pass


class TransactionRead(TransactionBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: datetime
    updated_at: datetime
